jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/TimerUpdateTest.java

```diff
@@ -0,0 +1,309 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.test.functional.timer;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.jbpm.process.instance.command.UpdateTimerCommand;
+import org.jbpm.test.JbpmTestCase;
+import org.jbpm.test.listener.CountDownProcessEventListener;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.kie.api.event.process.DefaultProcessEventListener;
+import org.kie.api.event.process.ProcessStartedEvent;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Test for UpdateTimmerCommand
+ */
+public class TimerUpdateTest extends JbpmTestCase {
+
+    private static final Logger logger = LoggerFactory.getLogger(TimerPersistenceTest.class);
+
+    private static final String TIMER_FILE = "org/jbpm/test/functional/timer/UpdateTimer.bpmn2";
+    private static final String PROCESS_NAME = "UpdateTimer";
+    private static final String TIMER_NAME = "Timer";
+
+    private static final String START_TIMER_FILE = "org/jbpm/test/functional/timer/UpdateStartTimer.bpmn2";
+    private static final String START_PROCESS_NAME = "UpdateStartTimer";
+    private static final String START_TIMER_NAME = "StartTimer";
+
+    private static final String BOUNDARY_TIMER_FILE = "org/jbpm/test/functional/timer/UpdateBoundaryTimer.bpmn2";
+    private static final String BOUNDARY_PROCESS_NAME = "UpdateBoundaryTimer";
+    private static final String BOUNDARY_TIMER_NAME = "BoundaryTimer";
+
+    private RuntimeEngine runtimeEngine;
+    private KieSession kieSession;
+
+    private static final String TIMER_FIRED_TEXT = "Hello Adele";
+    private static final String TIMER_FIRED_TIME_PROP = "Time Adele";
+
+    @Before
+    public void setup() {
+        runtimeEngine = null;
+        kieSession = null;
+        System.clearProperty(TIMER_FIRED_TEXT);
+        System.clearProperty(TIMER_FIRED_TIME_PROP);
+    }
+
+    @Test(timeout = 30000)
+    public void updateTimerLongerDelayTest() {
+        CountDownProcessEventListener countDownListener = new CountDownProcessEventListener(TIMER_NAME, 1);
+        //delay is set for 5s
+        setProcessScenario(TIMER_FILE);
+
+        kieSession.addEventListener(countDownListener);
+        final List<Long> list = new ArrayList<Long>();
+        kieSession.addEventListener(new DefaultProcessEventListener() {
+            @Override
+            public void beforeProcessStarted(ProcessStartedEvent event) {
+                list.add(event.getProcessInstance().getId());
+            }
+        });
+
+        assertEquals(0, list.size());
```

Comment:
I believe that the project already uses AssertJ in places. Unless the others mind, I'd like this to be ```Assertions.assertThat(list).isEmpty()```. Much clearer in the code, and a very descriptive error message when it fails.

@mswiderski Do you agree?

(Ditto with every other assertion.)